File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/di/module/SensorModule.java
package pl.tobzzo.miechowskiepowietrze.di.module;

import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import pl.tobzzo.miechowskiepowietrze.database.DataBaseManager;
import pl.tobzzo.miechowskiepowietrze.database.MpowDataBaseManager;

@Module
public class SensorModule {
  @Provides
  //@Singleton
  public DataBaseManager provideDataBaseManager() { return new MpowDataBaseManager(); }
}

File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/di/SensorModule.kt
package pl.tobzzo.miechowskiepowietrze.di


import dagger.Binds
import dagger.Module
import pl.tobzzo.miechowskiepowietrze.example.ExampleInterface
import pl.tobzzo.miechowskiepowietrze.example.MpowExampleImplementation
import pl.tobzzo.miechowskiepowietrze.network.MpowNetworkComponent
import pl.tobzzo.miechowskiepowietrze.network.NetworkComponent
import javax.inject.Singleton

@Module
abstract class SensorModule {

  @Binds
  internal abstract fun bindExample(example: MpowExampleImplementation): ExampleInterface
}
